src/shared_modules/indexer_connector/tests/component/fakeIndexer.hpp

```diff
@@ -0,0 +1,209 @@
+/*
+ * Wazuh Indexer Connector - Component tests
+ * Copyright (C) 2015, Wazuh Inc.
+ * January 09, 2024.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _FAKE_OPEN_SEARCH_SERVER_HPP
+#define _FAKE_OPEN_SEARCH_SERVER_HPP
+
+#include <external/cpp-httplib/httplib.h>
+#include <external/nlohmann/json.hpp>
+#include <functional>
+#include <sstream>
+#include <string>
+#include <thread>
+#include <utility>
+
+/**
+ * @brief This class is a simple HTTP server that provides a fake OpenSearch server.
+ */
+class FakeIndexer
+{
+private:
+    httplib::Server m_server;
+    std::thread m_thread;
+    int m_port;
+    std::string m_host, m_health, m_indexName;
+    bool m_indexerInitialized;
+    std::function<void(const std::string&)> m_initTemplateCallback, m_initIndexCallback, m_publishCallback;
+
+public:
+    /**
+     * @brief Class constructor.
+     *
+     * @param host Host of the server.
+     * @param port Port of the server.
+     * @param health Health status of the server.
+     * @param indexName Name of the index.
+     */
+    FakeIndexer(std::string host, int port, std::string health, std::string indexName)
+        : m_thread(&FakeIndexer::run, this)
+        , m_host(std::move(host))
+        , m_health(std::move(health))
+        , m_port(port)
+        , m_indexName(std::move(indexName))
+        , m_indexerInitialized(false)
+        , m_publishCallback({})
+        , m_initIndexCallback({})
+        , m_initTemplateCallback({})
```

Comment:
Suggestion: 
These could use in-class initialization ([Prefer in-class initializer](https://rules.sonarsource.com/cpp/RSPEC-3230/))

Reply:
Done.

Reply:
Already initialized in-class
```suggestion
        , m_indexerInitialized(false)
```

Reply:
Yes, fixed.